[Eric Lee Ying Yao] iP

autumn-sonata opened the pull request:
## DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

autumn-sonata committed:
Added greeting message

autumn-sonata committed:
Echoes user response and exits when 'bye' entered

autumn-sonata committed:
Modified greeting message

autumn-sonata committed:
Added storing and displaying of past texts

autumn-sonata committed:
Added indentation and reply formatting

autumn-sonata committed:
Added task class

autumn-sonata committed:
Added marked slot in list command

autumn-sonata committed:
Added checked exceptions with recovery

autumn-sonata committed:
Added getStatusIcon method

autumn-sonata committed:
Added deadline, event, todo files

autumn-sonata committed:
Added deadline, event, todo commands

autumn-sonata committed:
Added access control to Task.java

autumn-sonata committed:
Added text UI testing to program

autumn-sonata committed:
Added mark test case

autumn-sonata committed:
Added Mark Exception classes

autumn-sonata committed:
Added exceptions for deadline, event, todo

autumn-sonata committed:
Added UnrecognisedCommandException

autumn-sonata committed:
Fixed bug; commands may now be unrecognised

autumn-sonata committed:
Added deletion functionality of tasks

autumn-sonata committed:
Merge pull request #1 from autumn-sonata/Week-1

Week 1

autumn-sonata committed:
Add storage of data across instances

autumn-sonata committed:
Add proper datetime format to deadline and event

autumn-sonata committed:
Merge branch 'branch-Level-7'

autumn-sonata committed:
Merge branch 'branch-Level-8'

autumn-sonata committed:
Refactor FileIO into Storage class

autumn-sonata committed:
Add skeletal UI class

autumn-sonata committed:
Modify output to be through UI class

autumn-sonata committed:
Refactor userTasks into TaskList

autumn-sonata committed:
Refactor Parser and read of user commands

autumn-sonata committed:
Merge tag 'A-MoreOOP'

autumn-sonata committed:
Add packaging to program

autumn-sonata committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

autumn-sonata committed:
Add Gradle to project

autumn-sonata committed:
Add testing for TaskList and Todo classes

autumn-sonata committed:
Add javadocs to all classes

autumn-sonata committed:
Apply coding standard to project

autumn-sonata committed:
Add skeletal version of FindCmd class

autumn-sonata committed:
Add remaining Javadocs in TaskList

autumn-sonata committed:
Add find command

autumn-sonata committed:
Rectify bugs from test cases

autumn-sonata committed:
Removed emojis due to loading errors

autumn-sonata committed:
Change UI to Ui to adhere to coding standards

autumn-sonata committed:
Add additional UI to Ui naming standard changes

autumn-sonata committed:
Add Javadocs for TaskList class

autumn-sonata committed:
Change UI class to Ui to match coding standards

autumn-sonata committed:
Remove emojis that cannot be properly formatted

autumn-sonata committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/drivers/TaskList.java

autumn-sonata committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/commands/EndChatBotCmd.java

autumn-sonata committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/drivers/Ui.java

autumn-sonata committed:
Resolve Ui class renaming

autumn-sonata committed:
Modify based on checkstyle

chm780 commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,43 @@
+import exceptions.TumuException;
+import tasks.Task;
+
+/**
```
Good that you have javadocs for classes :)

chm780 commented on src/main/java/commands/DeadlineCmd.java:
```diff
@@ -0,0 +1,55 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws TumuException {
+        //Check for "/by", if not available then prompt user to add timing.
```
Good to let the code readers know the logic of code 👍

chm780 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,133 @@
-| | | | | | | |/ / _ \
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
+	 ▄▄▄▄▄▄▄ ▄▄   ▄▄ ▄▄   ▄▄ ▄▄   ▄▄ 
```
WOW nice!

chm780 commented on src/main/java/drivers/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+     */
+    public Task deleteTask(int oneIndexedNum) throws TumuException {
+        if (oneIndexedNum < 1 || oneIndexedNum > userTasks.size()) {
+            //Specified index from user is out of bounds of list.
```
Nice of you to guide the code reader :)

chm780 reviewed (commented):
Very pleasant code to read overall

autumn-sonata committed:
Move output from command line to JavaFX interface

autumn-sonata committed:
Add greeting message to interface

autumn-sonata committed:
Add application close on bye

autumn-sonata committed:
Merge branch 'branch-Level-10'

autumn-sonata committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	src/main/java/drivers/Tumu.java
#	src/main/java/drivers/Ui.java
#	src/main/java/exceptions/DeadlineEventTimingOverflowException.java
#	src/main/java/exceptions/UnrecognisedCommandException.java

ShenyiCui commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Executes the command and gives the appropriate
+     * feedback to the user.
+     * @param tasks TaskList containing all the tasks currently available.
```
You can add a line after your method description for better readability.

ShenyiCui commented on src/main/java/commands/DeadlineCmd.java:
```diff
@@ -0,0 +1,53 @@
+            throw new DeadlineEventNoTimingException("by");
+        } else {
+            //Parse the string. Make sure there is no multiple "/by" statements.
+            String[] parse = body.split("/by");
```
Good code shouldn't require comments for example, consider the following instead:
```Java
String[] parse = body.split("/by");
Boolean isEventTimingOverflow = parse.length > 2;

if (isEventTimingOverflow) {
    throw new ...
}
```
your variable naming needs work

ShenyiCui commented on src/main/java/commands/FindCmd.java:
```diff
@@ -0,0 +1,47 @@
+ *
+ * Finds all entries that contain the keyword inside the task description.
+ */
+public class FindCmd extends Command {
```
I would spell out the full class, `FindCommand`. Try to avoid unnecessary acronyms when you can.

ShenyiCui commented on src/main/java/commands/TodoCmd.java:
```diff
@@ -0,0 +1,41 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws TumuException {
+        if (body.isBlank()) {
+            throw new TodoException();
+        } else {
```
Redundant else statement. 
Try this instead:
```Java
if (body.isBlank()) {
    throw new TodoException();
}
return addTaskType(new Todo(body), storage, tasks, ui);
```
Neater and more readable this way

ShenyiCui commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+    public Deadline(String taskDescription, String timing) throws UnrecognisedDateTimeException {
+        super(taskDescription);
+        try {
+            //Alter the timing so that it can be properly added in.
```
add a space after your comment
`// Alter`... not `//Alter`

ShenyiCui reviewed (changes requested):
Very good detailed code! just some changes

autumn-sonata committed:
Add assertions to check for integer bounds and null objects.

The program currently runs on the assumption that certain objects will
never be null, or that certain indexes will be within the limit of the
number of tasks.

Assertions are assumptions that are made which, if broken, causes
the program to not run as intended. Assertions are implemented to make
sure that it is easier to debug the program in future changes.

Assertions are inserted right before points in the program where it is
critical that the assumption listed is correct. This includes the check
on:
* Whether there is no referenced task in Command.java
* Indexes to delete, mark and unmark tasks in the list

Assertions are placed directly before usage to prevent incorrect
assumptions from happening should new code modify the checked variables
afterwards. This prevents an incorrect assertion from being inserted.

More assertions to be added in the future.

autumn-sonata committed:
Modify class names and comments to improve readability

The command class names use a short form (Cmd for command).
Additionally, comments are not helpful since they describe what the
program is doing.

Not everyone will understand what Cmd actually means, since it is not
a short form that is widely recognized by everyone. Comments should
not describe a program, but rather, aid other readers of the code to
understand the program better.

The command class names will change Cmd -> Command, so that the full
name is spelt out. Comments which are redundant and descriptive are
removed, or rephrased to help the reader understand the program.

This will improve overall readability of the program, which is one of
the main goals of programming both in a team and personal environment.

autumn-sonata committed:
Merge pull request #2 from autumn-sonata/branch-A-CodeQuality

Merge branch-A-CodeQuality

autumn-sonata committed:
Merge branch 'master' into branch-A-Assertions

autumn-sonata committed:
Merge pull request #3 from autumn-sonata/branch-A-Assertions

Add assertions to check for integer bounds and null objects.

autumn-sonata committed:
Add sorting of tasks based on chronological order

autumn-sonata committed:
Merge pull request #4 from autumn-sonata/branch-BCD-extension

Add sorting of tasks based on chronological order

autumn-sonata committed:
Add yml file for Continuous Integration feature

autumn-sonata committed:
Sequence import commands to align with convention

autumn-sonata committed:
Merge pull request #5 from autumn-sonata/branch-CI

Add yml file for Continuous Integration feature

autumn-sonata committed:
Add representative screenshot of product

autumn-sonata committed:
Add user guide to program

autumn-sonata committed:
Merge branch 'branch-A-UserGuide'

autumn-sonata committed:
Add spacing for header recognition in README

autumn-sonata committed:
Enable jar on multiple devices

autumn-sonata committed:
Fix phrasing of overview statements

autumn-sonata committed:
Remove unnecessary Javafx functions in tumu driver
